lib/irb/default_commands.rb

```diff
@@ -236,6 +229,27 @@ def self.load_command(command)
       nil
     end
 
+    def self.install_helper_methods
+      HelperMethod.helper_methods.each do |name, helper_method_class|
+        define_method name do |*args, **opts, &block|
+          verbose, $VERBOSE = $VERBOSE, nil
+          helper_ivar = "@_helper_method_#{name}".to_sym
+          helper = instance_variable_get(helper_ivar)
+
+          unless helper
+            helper = helper_method_class.new
+            instance_variable_set(helper_ivar, helper)
```

Comment:
Command execution uses `new(irb_context).execute(arg)` and does not cache the instance.
I think HelperMethod don't need to cache too, or should cache in another place to avoid adding instance variables to `workspace.main`
```ruby
From: a.rb @ line 2 :

    1: class A
 => 2:   def f = binding.irb
    3: end

irb(#<A:0x000000013019fb00>):001> conf;context;irb_context;
irb(#<A:0x000000013019fb00>):002> self
=> 
#<A:0x000000013019fb00
 @_helper_method_conf=#<IRB::HelperMethod::Conf:0x0000000127985878>,
 @_helper_method_context=#<IRB::HelperMethod::Context:0x0000000127985670>,
 @_helper_method_irb_context=#<IRB::HelperMethod::IrbContext:0x00000001279854b8>>
```